File: 796/solution1/solution1.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:

        if len(s) != len(goal):
            return False

        if len(s) >= 1 and len(goal) <= 100:
            
            new_str = s[1:] + s[0]
            while new_str != s:
                if new_str == goal:
                    return True
                new_str = new_str[1:] + new_str[0]

        return False
        

File: 796/solution1/solution1.cpp
#include <string>

class Solution {
public:
    bool rotateString(std::string s, std::string goal) {

        if (s.length() != goal.length()) {
            return false;
        }
        for (int i = 0; i < s.length(); i++) {
            if (s == goal) return true;
            std::rotate(s.begin(), s.begin()+1, s.end()); 
        }

        return false;        
    }
};